lib/Clustering.js

```diff
@@ -0,0 +1,263 @@
+            return this._exitCb(this);
+        }
+        return process.exit(this.getStatus());
+    }
+
+    /**
+     * Function called when a worker exited
+     *
+     * @param {Cluster.worker} worker - Current worker
+     * @param {number} i - Worker index
+     * @param {number} code - Exit code
+     * @param {string} signal - Exit signal
+     * @return {undefined}
+     */
+    _workerExited(worker, i, code, signal) {
+        // If the worker:
+        //   - was killed by a signal
+        //   - return an error code
+        //   - or just stopped
+        if (signal) {
+            this._logger.info('Worker killed by signal', {
+                signal,
+                id: i,
+                childPid: worker.process.pid,
+            });
+            this._workersStatus[i] = signal;
+        } else if (code !== 0) {
+            this._logger.error('Worker exit with code', {
+                code,
+                id: i,
+                childPid: worker.process.pid,
+            });
+            this._workersStatus[i] = code;
+        } else {
+            this._logger.info('Worker shutdown gracefully', {
+                id: i,
+                childPid: worker.process.pid,
+            });
+            this._workersStatus[i] = undefined;
+        }
+        this._workers[i] = undefined;
+        if (this._workersTimeout[i]) {
+            clearTimeout(this._workersTimeout[i]);
+            this._workersTimeout[i] = undefined;
+        }
+        // If we don't trigger the stop method, the watchdog
+        // will autorestart the worker
+        if (this._shutdown === false) {
+            return process.nextTick(() => this.startWorker(i));
+        }
+        // Check if an worker is still running
+        if (!this._workers.every(cur => cur === undefined)) {
+            return undefined;
+        }
+        return this._afterStop();
+    }
+
+    /**
+     * Function to start a worker
+     *
+     * @param {number} i Index of the starting worker
+     * @return {undefined}
+     */
+    startWorker(i) {
+        if (!cluster.isMaster) {
+            return;
+        }
+        // Fork a new worker
+        this._workers[i] = cluster.fork();
+        // Listen for message from the worker
+        this._workers[i].on('message', msg => {
+            // If the worker is ready, send him his id
+            if (msg === 'ready') {
+                this._workers[i].send({ msg: 'setup', id: i });
+            }
+        });
+        this._workers[i].on('exit', (code, signal) =>
+            this._workerExited(this._workers[i], i, code, signal));
+        // Trigger when the worker was started
+        this._workers[i].on('online', () => {
+            this._logger.info('Worker started', {
+                id: i,
+                childPid: this._workers[i].process.pid,
+            });
+        });
+    }
+
+    /**
+     * Function to put handler on cluster exit
+     *
+     * @param {function} cb - Callback when cluster exit
```

Comment:
It would be great to know what parameters are sent to this cb. (`this` and the signal)